frank/fit.py

```diff
@@ -0,0 +1,275 @@
+
+    print("""
+     Fit a 1D radial brightness profile with Frankenstein (frank) from the
+     terminal with `python -m frank.fit`. A .json parameter file is required;
+     the default is default_parameters.json and is of the form:\n\n""",
+     json.dumps(param_descrip, indent=4))
+
+
+def parse_parameters():
+    """
+    Read in a .json parameter file to set the fit parameters.
+
+    Parameters
+    ----------
+    parameter_filename : string
+            Parameter file (.json; see frank.fit.helper).
+            Defaults to `default_parameters.json`
+    uvtable_filename : string
+            UVTable file with data to be fit (ASCII, .npy or .npz). The UVTable
+            column format should be u [lambda]  v [lambda] Re(V) [Jy]
+            Im(V) [Jy] Weight [Jy^-2] # TODO: update if accept dfft formats
+
+    Returns
+    -------
+    model : dict
+            Dictionary containing model parameters the fit uses
+    """
+
+    import argparse
+
+    parser = argparse.ArgumentParser("Run a Frank fit, by default using"
+                                     " parameters in default_parameters.json")
+    parser.add_argument("-p", "--parameter_filename",
+                        default='default_parameters.json', type=str,
+                        help="Parameter file (.json; see frank.fit.helper)")
+    parser.add_argument("-uv", "--uvtable_filename", default=None, type=str,
+                        help="UVTable file with data to be fit. See"
+                             " frank.io.load_uvtable")
+
+    args = parser.parse_args()
+    model = json.load(open(args.parameter_filename, 'r'))
+
+    if args.uvtable_filename:
+        model['input_output']['uvtable_filename'] = args.uvtable_filename
+
+    if ('uvtable_filename' not in model['input_output'] or
+        not model['input_output']['uvtable_filename']):
+        raise ValueError("    uvtable_filename isn't specified."
+                 " Set it in the parameter file or run frank with"
+                 " python -m frank.fit -uv <uvtable_filename>")
+
+    if not model['input_output']['load_dir']:
+        model['input_output']['load_dir'] = os.getcwd()
+
+    if not model['input_output']['save_dir']:
+        model['input_output']['save_dir'] = model['input_output']['load_dir']
+
+    logging.basicConfig(level=logging.INFO,
+        format='%(message)s',
+        handlers=[
+        logging.FileHandler(model['input_output']['save_dir'] +
+        '/frank_fit.log', mode='w'), logging.StreamHandler()]
+        )
+
+    logging.info('\nRunning frank on %s'
+                 %model['input_output']['uvtable_filename'])
+
+    logging.info('  Saving parameters to be used in fit to'
+                 ' `frank_used_pars.json`')
+    with open('frank_used_pars.json', 'w') as f:
+        json.dump(model, f, indent=4)
+
+    return model
+
+
+def load_data(data_file):
+    """
+    Read in a UVTable with data to be fit. See frank.io.load_uvtable
```

Comment:
Doc needs Parameters / Returns

Reply:
Done.

> I've specified a few changes. In addition:
> 
> * [x]  Remove all .DS_Stores files
> * [ ]  Remove AS209_continuum.txt
> * [ ]  Check that all doc strings have "Parameters" and "Returns" not "Parameter" and "Return"
Where did you see this? I haven't proofread the codebase beyond my scripts yet.
> * [ ]  Complete IO.py
> * [ ]  Complete plot.py
> * [ ]  Fix the jupyter notebooks. At the moment I cannot run them
> * [ ]  Add tests for I/O
> * [ ]  Add expliciit `git clone` followed by `pip install .`  info to installation.
> * [ ]  Add UVTable extraction page to docs
> * [ ]  Add figures to the docs
> 
> Also, what are the docs/_*/.placeholder files for? They are empty
I don't know, they were there be default or MT added. I've removed.

Reply:
Done.